aioshelly/coap.py

```diff
@@ -26,7 +26,7 @@ def __init__(self, sender_addr, payload: bytes):
             try:
                 self.payload = json.loads(payload.rsplit(b"\xff", 1)[1].decode())
             except (json.decoder.JSONDecodeError, IndexError):
-                _LOGGER.error(
+                _LOGGER.debug(
```

Comment:
It's weird that we set payload to None instead of raising this error if we receive invalid JSON. Now we need to guard for payload = None later to detect if this went wrong. Shouldn't we just throw an `InvalidMessage` error?

Reply:
We indeed guard for it right after parsing:
https://github.com/home-assistant-libs/aioshelly/blob/4ea230e5f88e8dbaac6ddb3d2e601345644982e8/aioshelly/coap.py#L91

I agree that raising an `InvalidMessage` is a better approach but than we need to rewrite this function to change this also:
https://github.com/home-assistant-libs/aioshelly/blob/4ea230e5f88e8dbaac6ddb3d2e601345644982e8/aioshelly/coap.py#L37

Do you think it is better to rewrite it and catch all errors here:
https://github.com/home-assistant-libs/aioshelly/blob/4ea230e5f88e8dbaac6ddb3d2e601345644982e8/aioshelly/coap.py#L88

So that we can raise errors if the IP is in the list or silently log it if it is not?

Reply:
Yeah that sounds better.